### DISCUSSION TITLE
【FormDialog】FormDialog的onCancel对antD的oncancel进行了重写，导致无法拦截到关闭前的状态

### DISCUSSION THREAD

**Discussion by aoilti** [alibaba/formily#1723] [Ideas]
重写后的onCancal导致无法拦截到关闭modal事件，也就无法做类似于关闭前确认的功能，是不是可以增加beforeClose事件回调，根据回调返回值对是否关闭dialog做判断？
![image](https://user-images.githubusercontent.com/19969080/124554258-5ccf8100-de68-11eb-9342-0ad8b982979c.png)

**Comment by janryWang** [comment]
FormDialog设计上是有点不太完备，后面有空得再想一下

**Reply by aoilti** [reply]
要不先bugfix一下？我现在如果想做拦截，就需要在我传入的oncancel里throw一个error来打断后续代码执行，这样的方案不太优雅。可以先fix的话，你提方案，我可以近期提pr

**Reply by janryWang** [reply]
你先提issue，在discussion里是没法跟踪的

**Reply by liuweiGL** [reply]
> 要不先bugfix一下？我现在如果想做拦截，就需要在我传入的oncancel里throw一个error来打断后续代码执行，这样的方案不太优雅。可以先fix的话，你提方案，我可以近期提pr

在 onCancel 里面判断一下 `modal?.onCancel?.(e) === false` 吧。

**Reply by janryWang** [reply]
嗯，我也是这样想的，这个问题是可以这样解决的

**Reply by aoilti** [reply]
好的
